rusk-wallet/src/gql.rs

```diff
@@ -58,14 +61,50 @@ struct BlockResponse {
     pub block: Option<Block>,
 }
 
+// See `PhoenixTransactionEvent` for the reason for this struct
+// and allowing dead code here.
 #[serde_as]
-#[derive(Deserialize, Debug)]
-pub struct BlockData {
-    #[serde_as(as = "DisplayFromStr")]
-    pub gas_spent: u64,
-    pub sender: String,
+#[derive(Deserialize, Debug, Clone)]
+#[allow(dead_code)]
+struct Note {
+    stealth_address: StealthAddress,
+}
+
+// This struct is used instead of the one in `dusk_core::transfer`
+// because of the order-dependent deserialization bug in
+// the `dusk_core::transfer::phoenix::Note` https://github.com/dusk-network/phoenix/issues/274.
+// Dead code is allowed to avoid catch-alls, so that the case in which an
+// unexpected event is received, an appropriate error will be thrown.
+#[serde_as]
+#[derive(serde::Deserialize, Debug)]
+#[allow(dead_code)]
+pub struct PhoenixTransactionEvent {
+    /// Notes produced during the transaction.
+    notes: Vec<Note>,
+    /// The memo included in the transaction.
+    #[serde_as(as = "Hex")]
+    memo: Vec<u8>,
+    /// Gas spent by the transaction.
     #[serde_as(as = "DisplayFromStr")]
-    pub value: f64,
+    gas_spent: u64,
+    /// Optional gas-refund note if the refund is positive.
+    refund_note: Option<Note>,
+}
+
+/// Deserialized block data in the full moonlight history.
+#[serde_as]
+#[derive(Deserialize, Debug)]
+#[serde(untagged)]
+#[allow(clippy::large_enum_variant)]
+pub enum BlockData {
+    /// For the moonlight transaction event.
+    MoonlightTransactionEvent(MoonlightTransactionEvent),
+    /// For the PhoenixTransactionEvent.
+    /// In the case where a conversion is made from phoenix to
+    /// moonlight, this appears.
+    PhoenixTransactionEvent(PhoenixTransactionEvent),
+    /// For the convert event.
+    ConvertEvent(ConvertEvent),
```

Comment:
you use `BlockEvent` as a wrapper for `BlockData`. Why introduce a new type and not modify `BlockEvent` directly?

Reply:
I did not introduce this type - I modified it.
`BlockEvents` seems to have been put there because of the JSON deserialization of the full moonlight history.

Reply:
Can they be collapsed into one?

Reply:
I'm not sure they can.